### DISCUSSION TITLE
Feature request: `skip_as_script` should propagate to cell dependencies

### DISCUSSION THREAD

**Discussion by greimel** [fonsp/Pluto.jl#2175] [Pluto development] [answered]
I am really excited that #2018 got merged. This functionality will be really useful! (Finally no more typing `@skip_as_script` – I have always mixed up the `c` and the `k`.)

As I [commented in the PR](https://github.com/fonsp/Pluto.jl/pull/2018#issuecomment-1152430038) I think it would be even more useful if `skip_as_script` propagated to dependent cells. Here's why.

1. The current implementation makes it quite easy to have a notebook that is not runnable as a script (thus breaking a main selling point of Pluto).
2. One major use case of this feature would be for writing documentation. You _skip_ the cell with `using Plots` or the cell that creates your test data, and all the plots and illustrations are automatically skipped when `including` the file in the package.

As an aside, this would make it easier to have course notebooks that depends on each other. Because usually these are heavy on plots and simulations, besides the reusable core functionality.

**Comment by fonsp** [comment, marked as answer]
EDIT: This has been implemented in #2178 !

Thanks for the suggestion!

What would be the disadvantages of this? @dralletje @disberd I think we discussed this before?

**Reply by disberd** [reply]
I am not sure we discussed about this before but I think it would indeed make a lot of sense functionality wise.
I didn't look at the code yet but I believe we could mostly reuse the code for tracking cells depending on disabled cells, but indeed probably it would be good to start looking at an implementation to better find potential problems!

I can try having a look in the next days to see if I can make a draft PR to add the functionality.

**Reply by fonsp** [reply]
I was about to do a Pluto release that will include the feature in the current state. Should I put it behind an experimental flag until we figure this out?

**Reply by disberd** [reply]
I would put the current state in a release (as it's already useful as it is and by exposing it we may find potential issues) and then go for advanced functionality in a subsequent release after the PR

**Reply by disberd** [reply]
I also believe that the current state is the one that at least in my opinion is sure to be useful and most likely to be feasible without any side effects, so I am for going for it without the flag (but maybe I am biased towards this functionality :D)

**Reply by fonsp** [reply]
Hmmm I put it behind a flag because implementing this suggestion by Fabian will change the behaviour of notebooks.

https://github.com/fonsp/Pluto.jl/commit/4e7a48d7087036f40d03e88f51bb830fa90a1516?w=1

**Reply by disberd** [reply]
How are these flags accessed? do I have to clone the repo, change the flag in the source and develop the package to have it work?

Edit: OK I have seen the commit and it seems I just have to call `window.PLUTO_TOGGLE_SKIP_AS_SCRIPT()` in javascript to toggle.
